jsdependencies-sbt-plugin/src/main/scala/org/scalajs/jsdependencies/sbtplugin/JSDependenciesPlugin.scala

```diff
@@ -0,0 +1,364 @@
+package org.scalajs.jsdependencies.sbtplugin
+
+import scala.collection.mutable
+import scala.util.Try
+
+import java.io.InputStreamReader
+
+import sbt._
+import sbt.Keys._
+
+import org.scalajs.core.ir.Utils.escapeJS
+
+import org.scalajs.core.tools.io.{IO => toolsIO, _}
+import org.scalajs.core.tools.json._
+
+import org.scalajs.jsenv.VirtualFileMaterializer
+
+import org.scalajs.sbtplugin.ScalaJSPlugin
+import org.scalajs.sbtplugin.ScalaJSPlugin.autoImport._
+import org.scalajs.sbtplugin.ScalaJSPluginInternal.scalaJSSourceFiles
+
+import org.scalajs.jsdependencies.core._
+import org.scalajs.jsdependencies.core.DependencyResolver.DependencyFilter
+import org.scalajs.jsdependencies.core.ManifestFilters.ManifestFilter
+
+object JSDependenciesPlugin extends AutoPlugin {
+  override def requires: Plugins = ScalaJSPlugin
+
+  object autoImport {
+    import KeyRanks._
+
+    val scalaJSNativeLibraries = TaskKey[Attributed[Seq[VirtualJSFile with RelativeVirtualFile]]](
+        "scalaJSNativeLibraries", "All the *.js files on the classpath", CTask)
+
+    val packageJSDependencies = TaskKey[File]("packageJSDependencies",
+        "Packages all dependencies of the preLink classpath in a single file.", AMinusTask)
+
+    val packageMinifiedJSDependencies = TaskKey[File]("packageMinifiedJSDependencies",
+        "Packages minified version (if available) of dependencies of the preLink " +
+        "classpath in a single file.", AMinusTask)
+
+    val jsDependencyManifest = TaskKey[File]("jsDependencyManifest",
+        "Writes the JS_DEPENDENCIES file.", DTask)
+
+    val jsDependencyManifests = TaskKey[Attributed[Traversable[JSDependencyManifest]]](
+        "jsDependencyManifests", "All the JS_DEPENDENCIES on the classpath", DTask)
+
+    val requiresDOM = SettingKey[Boolean]("requiresDOM",
+        "Whether this projects needs the DOM. Overrides anything inherited through dependencies.", AMinusSetting)
```

Comment:
This setting is unused (short of `scalaJSRequestsDOM`). Same story about dropping than for `scalaJSRequestsDOM`.